### ISSUE TITLE
redefined-argument-from-local false positive with lambda in argument list

### ISSUE THREAD

**Issue by mthuurne** [PyCQA/pylint#2364]
### Steps to reproduce

Run pylint on the following code:

```python
def apply_filter(objects, filt=lambda obj: True):
    for obj in objects:
        if filt(obj):
            yield obj
```

### Current behavior

A message is issued about redefining `obj`:
```
2:8: R1704: Redefining argument with the local name 'obj' (redefined-argument-from-local)
```

### Expected behavior

No message should be issued, since `obj` isn't an argument of `apply_filter`: it's an argument of the `lambda` and therefore its scope is limited to that `lambda`.

Note that the use of a `for` loop is necessary to trigger this false positive: if `obj` is just assigned to as a local variable, the problem does not occur. Maybe the `for` and the `lambda` accidentally share the same limited scope, instead of having their own scopes?

### pylint --version output

```
pylint 2.0.0
astroid 2.0.0.dev4
Python 3.4.6 (default, Mar 22 2017, 12:26:13) [GCC]
```
